StandaloneMmPkg/Core/Pool.c

```diff
@@ -63,6 +77,15 @@ MmInitializeMemoryServices (
       MmramRanges[Index].RegionState
       );
   }
+
+  if (FixedPcdGetBool (PcdRuntimeMemoryOnHeap)) {
+    MmAddMemoryRegion (
+      (UINTN)FreePoolOnHeap,
+      sizeof (FreePoolOnHeap),
+      EfiConventionalMemory,
+      EFI_CACHEABLE
+      );
+  }

```

Comment:
I think this is not fit for the plan to remove Hob creation in Arm StandaloneMm.
Therefore, Although it pasess to CreateHobList.c in Arm, StandaloneMm couldn't create Hob :(

Reply:
I think it's okay that the Standalone MM core modifies some of the HOB contents. The HOB read-only restriction applies to all MM drivers.

See https://github.com/tianocore/edk2/blob/0f9dbb4abf540b606d1319af04482fcf46c55f6f/StandaloneMmPkg/Core/StandaloneMmCore.c#L656.

Reply:
But, modification should be done after StandaloneMmCore call InitializeMmHobList 

(See https://github.com/tianocore/edk2/blob/0f9dbb4abf540b606d1319af04482fcf46c55f6f/StandaloneMmPkg/Core/StandaloneMmCore.c#L856)

Becuase StandaloneMmEntryPoint's Hob list passed in https://github.com/tianocore/edk2/pull/6116., Could be readonly memory from firmware.

So, I think it's reasonable to modify contents in here.

However, I'm not sure which platform whether there is the case of 
"The caller might not know how much memoryspace to allocate."

Since when StandaloneMm loaded, the Heap memory space is specified by manifest file and delivered always to StandaloneMm, I don't know it's useful for Arm.